src/collapse/ghcprof.rs

```diff
@@ -0,0 +1,222 @@
+        Ok(())
+    }
+
+    /// Check for start line of a call graph.
+    fn is_applicable(&mut self, input: &str) -> Option<bool> {
+        let mut input = input.as_bytes();
+        let mut line = String::new();
+        loop {
+            line.clear();
+            if let Ok(n) = input.read_line(&mut line) {
+                if n == 0 {
+                    break;
+                }
+            } else {
+                return Some(false);
+            }
+
+            if line
+                .split_whitespace()
+                .take(START_LINE.len())
+                .eq(START_LINE.iter().cloned())
+            {
+                return Some(true);
+            }
+        }
+        None
+    }
+}
+
+impl From<Options> for Folder {
+    fn from(opt: Options) -> Self {
+        Folder {
+            opt,
+            ..Default::default()
+        }
+    }
+}
+
+impl Folder {
+    // Handle call graph lines of the form:
+    //
+    // MAIN           MAIN ...
+    //  CAF           Options.Applicative.Builder ...
+    //   defaultPrefs Options.Applicative.Builder ...
+    //    idm         Options.Applicative.Builder ...
+    //    prefs       Options.Applicative.Builder ...
+    //   fullDesc     Options.Applicative.Builder ...
+    //   hidden       Options.Applicative.Builder ...
+    //   option       Options.Applicative.Builder ...
+    //    metavar     Options.Applicative.Builder ...
+    //  CAF           Options.Applicative.Builder.Internal ...
+    //   internal     Options.Applicative.Builder.Internal ...
+    //   noGlobal     Options.Applicative.Builder.Internal ...
+    //   optionMod    Options.Applicative.Builder.Internal ...
+
+    fn on_line(
+        &mut self,
+        line: &str,
+        occurrences: &mut Occurrences,
+        cols: &Cols,
+    ) -> io::Result<()> {
+        if let Some(indent_chars) = line.find(|c| c != ' ') {
+            let prev_len = self.stack.len();
+            let depth = indent_chars;
+
+            if depth < prev_len {
+                // If the line is not a child, pop stack to the stack before the new depth
+                self.stack.truncate(depth);
+            } else if depth != prev_len {
+                return invalid_data_error!("Skipped indentation level at line:\n{}", line);
+            }
+            // There can be non-ascii names so take care to char offset not byte offset
+            let string_range = |col_start: usize| {
+                line.chars()
+                    .skip(col_start)
+                    .skip_while(|c| c.is_whitespace())
+                    .take_while(|c| !c.is_whitespace())
+                    .collect::<String>()
+            };
+            let cost = string_range(cols.source);
+            if let Ok(cost) = cost.trim().parse::<f64>() {
+                let func = string_range(cols.cost_centre);
+                let file = string_range(cols.module);
+                // Costs include self + calls (at least in what we parse)
```

Comment:
I don't think this is true for `%time`. At least the _first_ `%time` column is listed as "individual", which makes me think it doesn't include child calls. May be true for ticks/bytes though?

Reply:
You're right. I've corrected the comment